subsys/bluetooth/mesh/beacon.c

```diff
@@ -502,6 +502,20 @@ static bool secure_beacon_authenticate(struct bt_mesh_subnet *sub, void *cb_data
 	for (int i = 0; i < ARRAY_SIZE(sub->keys); i++) {
 		if (sub->keys[i].valid && auth_match(&sub->keys[i], params)) {
 			params->new_key = (i > 0);
+#if defined(CONFIG_BT_TESTING)
+			struct bt_mesh_snb beacon_info;
+
+			STRUCT_SECTION_FOREACH(bt_mesh_beacon_cb, cb) {
+				if (cb->snb_received) {
+					beacon_info.flags = params->flags;
```

Comment:
Can `beacon_info` be filled in only once? It should not be changed since it is const.